support/upsert/upsert.go

```diff
@@ -173,14 +173,15 @@ func defaultServiceSpec(original, mutated *corev1.ServiceSpec) {
 		mutated.IPFamilyPolicy = original.IPFamilyPolicy
 	}
 	for i := range original.Ports {
-		if i >= len(mutated.Ports) {
-			break
-		}
-		if mutated.Ports[i].Protocol == "" {
-			mutated.Ports[i].Protocol = original.Ports[i].Protocol
-		}
-		if mutated.Ports[i].TargetPort.String() == "0" {
-			mutated.Ports[i].TargetPort = intstr.FromInt(int(mutated.Ports[i].Port))
+		for j := range mutated.Ports {
+			if original.Ports[i].Name == mutated.Ports[j].Name {
+				if mutated.Ports[j].Protocol == "" {
+					mutated.Ports[j].Protocol = original.Ports[i].Protocol
+				}
+				if mutated.Ports[j].TargetPort.String() == "0" {
+					mutated.Ports[j].TargetPort = intstr.FromInt(int(mutated.Ports[j].Port))
```

Comment:
I may be missing something here, but should this be from `mutated` to `mutated` ?